src/frontend/src/eth/services/eth-transactions-batch.services.ts

```diff
@@ -0,0 +1,51 @@
+import { ETHERSCAN_MAX_CALLS_PER_SECOND } from '$env/rest/etherscan.env';
+import { loadEthereumTransactions } from '$eth/services/eth-transactions.services';
+import { batch } from '$lib/services/batch.services';
+import type { Token, TokenId } from '$lib/types/token';
+import type { ResultSuccess } from '$lib/types/utils';
+
+export type ResultByToken = ResultSuccess & { tokenId: TokenId };
+
+type PromiseResult = Promise<ResultByToken>;
+
+export const batchLoadTransactions = ({
+	tokens,
+	tokensAlreadyLoaded
+}: {
+	tokens: Token[];
+	tokensAlreadyLoaded: TokenId[];
+}): AsyncGenerator<PromiseSettledResult<ResultByToken>[]> => {
+	const promises = tokens.reduce<PromiseResult[]>(
+		(acc, { network: { id: networkId }, id: tokenId }) => {
+			if (tokensAlreadyLoaded.includes(tokenId)) {
+				return acc;
+			}
+
+			const promise = (async (): PromiseResult => {
+				const result = await loadEthereumTransactions({ tokenId, networkId });
+				return { ...result, tokenId };
+			})();
+
+			return [...acc, promise];
+		},
+		[]
+	);
+
+	return batch<ResultByToken>({
+		promises,
+		batchSize: ETHERSCAN_MAX_CALLS_PER_SECOND
+	});
+};
+
+export const batchResultsToTokenId = (results: PromiseSettledResult<ResultByToken>[]): TokenId[] =>
+	results.reduce<TokenId[]>((acc, result) => {
+		if (result.status === 'rejected') {
+			return acc;
+		}
+
+		if (!result.value.success) {
```

Comment:
it is part of the results input, since they have the type

PromiseSettledResult<ResultByToken>

that in case of fulfilled promise is something like

```typescript
{
	status: 'fulfilled',
	value: { success, tokenId }
}
```